lib/srv/db/access_test.go

```diff
@@ -99,83 +99,130 @@ func TestMain(m *testing.M) {
 // on the configured RBAC rules.
 func TestAccessPostgres(t *testing.T) {
 	ctx := context.Background()
-	testCtx := setupTestContext(ctx, t, withSelfHostedPostgres("postgres"))
+	testCtx := setupTestContext(ctx, t, withSelfHostedPostgres("postgres", func(db *types.DatabaseV3) {
+		db.SetStaticLabels(map[string]string{"foo": "bar"})
+	}))
 	go testCtx.startHandlingConnections()
 
+	wildCardDBLabels := types.Labels{types.Wildcard: []string{types.Wildcard}}
+	dynamicDBLabels := types.Labels{"echo": {"test"}}
+	staticDBLabels := types.Labels{"foo": {"bar"}}
 	tests := []struct {
-		desc         string
-		user         string
-		role         string
-		allowDbNames []string
-		allowDbUsers []string
-		dbName       string
-		dbUser       string
-		err          string
+		desc          string
+		user          string
+		role          string
+		allowDbNames  []string
+		allowDbUsers  []string
+		allowDbLabels types.Labels
+		denyDbLabels  types.Labels
+		dbName        string
+		dbUser        string
+		err           string
 	}{
 		{
-			desc:         "has access to all database names and users",
-			user:         "alice",
-			role:         "admin",
-			allowDbNames: []string{types.Wildcard},
-			allowDbUsers: []string{types.Wildcard},
-			dbName:       "postgres",
-			dbUser:       "postgres",
+			desc:          "has access to all database names and users",
+			user:          "alice",
+			role:          "admin",
+			allowDbNames:  []string{types.Wildcard},
+			allowDbUsers:  []string{types.Wildcard},
+			allowDbLabels: wildCardDBLabels,
```

Comment:
Is it necessary to retrofit the labels in all test cases? Shouldn't we keep the existing scenarios unchanged?